ipaserver/install/installutils.py

```diff
@@ -156,17 +168,26 @@ def verify_fqdn(host_name, no_host_dns=False, local_hostname=True):
         # make sure that the host name meets the requirements in ipalib
         validate_hostname(host_name, maxlen=MAXHOSTNAMELEN)
     except ValueError as e:
-        raise BadHostError("Invalid hostname '%s', %s" % (host_name, unicode(e)))
+        raise BadHostError("Invalid hostname '%s', %s"
+                           % (host_name, unicode(e)))
 
     if local_hostname:
         try:
             logger.debug('Check if %s is a primary hostname for localhost',
                          host_name)
             ex_name = socket.gethostbyaddr(host_name)
             logger.debug('Primary hostname for localhost: %s', ex_name[0])
+            # if FQDN is not the first item on the /etc/hosts file
             if host_name != ex_name[0]:
-                raise HostLookupError("The host name %s does not match the primary host name %s. "\
-                        "Please check /etc/hosts or DNS name resolution" % (host_name, ex_name[0]))
+                print(ex_name)
+                if host_name in ex_name or [host_name] in ex_name:
```

Comment:
This line is hard to understand.

Please rewrite the assignment as ``ex_name, aliases = socket.gethostbyaddr(host_name)[:2]`` and then check if the hostname is either equal to ex_name or in the aliases list.

Reply:
thanks for the tip sir. :-)